cmd/resource.go

```diff
@@ -0,0 +1,155 @@
+	cmd := &cli.Command{
+		Use:   "resource",
+		Short: "Interact with data resource",
+	}
+	cmd.AddCommand(createResourceSubCommand(l, datastoreSpecsFs, datastoreRepo))
+	return cmd
+}
+
+func createResourceSubCommand(l log.Logger, datastoreSpecFs map[string]afero.Fs, datastoreRepo models.DatastoreRepo) *cli.Command {
+	return &cli.Command{
+		Use:     "create",
+		Short:   "Create a new resource",
+		Example: "optimus resource create",
+		RunE: func(cmd *cli.Command, args []string) error {
+			availableStorer := []string{}
+			for _, s := range datastoreRepo.GetAll() {
+				availableStorer = append(availableStorer, s.Name())
+			}
+			var storerName string
+			if err := survey.AskOne(&survey.Select{
+				Message: "Select supported datastores?",
+				Options: availableStorer,
+			}, &storerName); err != nil {
+				return err
+			}
+			repoFS, ok := datastoreSpecFs[storerName]
+			if !ok {
+				return fmt.Errorf("unregistered datastore, please use configuration file to set datastore path")
+			}
+
+			// find requested datastore
+			availableTypes := []string{}
+			datastore, _ := datastoreRepo.GetByName(storerName)
+			for dsType := range datastore.Types() {
+				availableTypes = append(availableTypes, dsType.String())
+			}
+			resourceSpecRepo := local.NewResourceSpecRepository(repoFS, datastore)
+
+			// find resource type
+			var resourceType string
+			if err := survey.AskOne(&survey.Select{
+				Message: "Select supported resource type?",
+				Options: availableTypes,
+			}, &resourceType); err != nil {
+				return err
+			}
+			typeController, _ := datastore.Types()[models.ResourceType(resourceType)]
+
+			// find directory to store spec
+			rwd, err := getWorkingDirectory(repoFS, "")
+			if err != nil {
+				return err
+			}
+			newDirName, err := getDirectoryName(rwd)
+			if err != nil {
+				return err
+			}
+
+			resourceDirectory := filepath.Join(rwd, newDirName)
+			resourceNameDefault := strings.ReplaceAll(strings.ReplaceAll(resourceDirectory, "/", "."), "\\", ".")
+
+			var qs = []*survey.Question{
+				{
+					Name: "name",
+					Prompt: &survey.Input{
+						Message: "What is the resource name?(should conform to selected resource type)",
+						Default: resourceNameDefault,
+					},
+					Validate: survey.ComposeValidators(validateNoSlash, survey.MinLength(3),
+						survey.MaxLength(1024), IsValidDatastoreSpec(typeController.Validator()),
+						IsResourceNameUnique(resourceSpecRepo)),
+				},
+			}
+			inputs := map[string]interface{}{}
+			if err := survey.Ask(qs, &inputs); err != nil {
+				return err
+			}
+			resourceName := inputs["name"].(string)
+
+			if err := resourceSpecRepo.SaveAt(models.ResourceSpec{
+				Version:   1,
+				Name:      resourceName,
+				Type:      models.ResourceType(resourceType),
+				Datastore: datastore,
+				Assets:    typeController.DefaultAssets(),
+			}, resourceDirectory); err != nil {
+				return err
+			}
+
+			l.Info(fmt.Sprintf("resource created successfully %s", resourceName))
```

Comment:
add color success and make first letter capital?

Reply:
Fixed it.